src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -275,9 +293,15 @@ - (void)writeMessage:(NSData *)message withErrorHandler:(void (^)())errorHandler
       }
     }
   };
-  [_wrappedCall startBatchWithOperations:@[[[GRPCOpSendMessage alloc] initWithMessage:message
-                                                                              handler:resumingHandler]]
-                            errorHandler:errorHandler];
+
+  GRPCOpSendMessage *op = [[GRPCOpSendMessage alloc] initWithMessage:message
+                                                             handler:resumingHandler];
+  if (!_unaryCall) {
+    [_wrappedCall startBatchWithOperations:@[op]
+                              errorHandler:errorHandler];
+  } else {
+    [_unaryOpBatch addObject:op];
```

Comment:
assert that we don't get custom errorHandler here, since we are throwing it away.